Add SCForm template + consistent form field styling

frankrowe opened the pull request:
UMC-37

frankrowe committed:
create reusable native SCForm component

frankrowe committed:
style improvements

frankrowe committed:
style updates

frankrowe committed:
version 3 beta 1

frankrowe committed:
modal

frankrowe committed:
fix slider color

frankrowe committed:
remove third party libs

frankrowe committed:
consistent styling + error messages

joshkendrick committed:
check and display form validations

joshkendrick committed:
copy the config for text based fields

frankrowe committed:
add formSubmittedError method

joshkendrick commented:
Based on what I've been doing for field validations, this seems to be working and can probably be merged

joshkendrick committed:
always copy the field schema into config

this is so the field schema is available for other things, for example, validation

joshkendrick committed:
fix and finish validation checks

joshkendrick committed:
pass form values and schema to validate

this should be a more correct way of validating fields, to jive with code review comments

frankrowe committed:
allow module to be used for web + native

frankrowe committed:
validation alert

frankrowe committed:
improve field styles

frankrowe committed:
version 3.0.0
